src/viam/sdk/CMakeLists.txt

```diff
@@ -17,8 +17,13 @@ add_library(viam-cpp-sdk::viamsdk ALIAS viamsdk)
 target_include_directories(viamsdk
   PUBLIC
     "$<BUILD_INTERFACE:$<TARGET_PROPERTY:viam-cpp-sdk::viamapi,INTERFACE_INCLUDE_DIRECTORIES>>"
+  PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/../.."
```

Comment:
This makes it so the compiler can see the generated `version_metadata.hpp` file when building. It intuitively feels wrong to me to include the build dir here (even privately), but this is the solution proposed by the [cmake documentation](https://cmake.org/cmake/help/latest/command/configure_file.html#example), so :shrug:. My discomfort is naive (I don't really know well enough to say concretely _why_ I don't like it other than it just gives off a bit of a code smell), so if anyone has thoughts or suggestions I'm all ears.

The `/../..` suffix is necessary to get the include to look right/consistent in `utils.hpp`, otherwise the include path would be inconsistent with every other SDK file.

Reply:
I think this is expected  for a generated file, but I will take a quick look to see if there is a more idiomatic way to express it.

Reply:
I pulled your branch to experiment. My findings are that 1) yes this is needed, but 2) there is a somewhat better way to write it, and 3) it should be moved elsewhere in this CMakeLists.txt file.

This particular call to `target_include_directories` is doing something tricky to ensure that the proto paths come first, for reasons that are outlined in the comment above.

Instead, you can add a variant of this line to the *other* `target_include_directories` call in this file, down around line 194 (new), and you can write it like this:

```
target_include_directories(...
    ...
    PRIVATE
        "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/../..>"
```

Now, one thing I'm trying to figure out is how we manage to already have what is effectively `"$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../..>"` in our include directories, which is what makes the build work at all against the non-generated sources. But it is clearly necessary to tell cmake that we want to look in our build directory for this generated file, and this is the correct way to do it (scoped private, using a build-time generator).

Reply:
Awesome thanks for digging! Just updated.